rescan_test.go

```diff
@@ -0,0 +1,680 @@
+package neutrino
+
+import (
+	"errors"
+	"os"
+	"reflect"
+	"runtime/pprof"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/chaincfg"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/rpcclient"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/btcsuite/btcutil"
+	"github.com/btcsuite/btcutil/gcs"
+	"github.com/btcsuite/btcutil/gcs/builder"
+	"github.com/btcsuite/btcwallet/waddrmgr"
+	"github.com/davecgh/go-spew/spew"
+	"github.com/lightninglabs/neutrino/blockntfns"
+)
+
+// mockChainSource is a mock implementation of the ChainSource interface that
```

Comment:
This commit is quite large.  Possible to add the test/harness in a separate commit, then follow up with a commit that fixes the actual error?

Reply:
Done.